### DISCUSSION TITLE
OpenTF support

### DISCUSSION THREAD

**Discussion by skyzyx** [jfrog/terraform-provider-artifactory#807] [Q&A] [answered]
https://opentf.org/announcement

We want to understand how this provider is affected by the OpenTF announcement, and what the plans for support are moving forward.

**Comment by alexhung** [comment]
@skyzyx We don't have any plan to change currently. Since OpenTF is compatible with current and future Terraform releases, all providers should work with OpenTF without any modification.

If and when there are features forking, we will determine our support roadmap then.

In case HashiCorp's BSL announcement and FAQ are not clear, all SDK and plugins remains under Mozilla license so there's no effect to any 3rd party providers.

**Comment by skyzyx** [comment]
Thank you for your answer.

From a technical perspective, I expect that it will be a no-op for the time being as OpenTF anticipates maintaining compat/interop for the foreseeable future. (However, OpenTF has stated that _some_ functionality may be added to OpenTF that is a _superset_ of what's in base Terraform. [[source](https://github.com/opentffoundation/roadmap/issues/24)])

From a process perspective:

1. If there is a bug in OpenTF that does not exist in Terraform (or vice-versa), will bug reports (as they pertain to this project) be treated equally on both platforms?

1. Will testing be performed with code targeting both platforms?

1. For OpenTF's potential future "superset" functionality, will that be treated as a first-class citizen?

**Comment by alexhung** [comment]
>1. If there is a bug in OpenTF that does not exist in Terraform (or vice-versa), will bug reports (as they pertain to this project) be treated equally on both platforms?

When OpenTF is released and ready to be used/integrated with, then we will need to determine the amount of effort required. Is it literally a 'drop-in' replacement for users and thus nothing required from provider developers? Or some small configuration to toggle between HashiCorp and OpenTF? All of these are unknown at this time, so we can't promise anything either. Of course we hope for the former and even latter and thus we can support both but that's pure hope and speculation at this point.

>2. Will testing be performed with code targeting both platforms?

Hopefully yes, but see answer to #1.

>3. For OpenTF's potential future "superset" functionality, will that be treated as a first-class citizen?

Yes in concept but it really depends on how that is handled by OpenTF at the implementation level.

**Comment by alexhung** [comment, marked as answer]
@skyzyx From my current perspective, this is the current plan:

1. Given OpenTF promise of backward compatibility, any bug reports will be treated the same. Obviously if we determine the bug is caused upstream in OpenTF (e.g. works in Terraform but not OpenTF), that'll not be fixed by JFrog team.
2. We will add OpenTF to our CI/CD pipeline so we have confidence that the release is verified for both platforms. The question of whether OpenTF will provider their own registry seems to be still opened. So that's TBD for our release process.

**Comment by skyzyx** [comment]
This is the information I'm looking for. Thank you.

I certainly don't expect you to resolve upstream bugs, but knowing that OpenTF part of your testing/validation process, and that you would file bug reports upstream when you encounter issues is the level of commitment from JFrog that I'm looking for at the moment. (Also noting any known compat/interop issues in the release notes would be helpful.)

We leverage a lot of tools in the Terraform ecosystem, from open-source tools, to SaaS providers who are firmly in the OpenTF camp due to being "competitive" with HashiCorp. I'm trying to understand not only the position of our internal legal team on the license change, but also the technical impact of OpenTF _existing_ and all of us having to do this dance.

Hoping for the best from all this, but pro-actively planning/communicating for the worst.

Your answers are greatly appreciated. Thank you.
